### PR TITLE
Recovery phrase chips onboarding unit tests

### PR DESCRIPTION
**Pull request by tmashuang**
## Explanation
Recovery phrase chips onboarding unit tests
Broken apart from #17360.
Partially completes #17191

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by darkwing — COMMENTED** [review]
Would it make sense to ensure the recovery phrase words, in that order, match the recovery phrase?  This is just counting elements.

**Comment by tmashuang** [comment]
@darkwing Thanks for the review! I forgot a test file/pr for the `review-recovery-phrase`, which I just made a pr for https://github.com/MetaMask/metamask-extension/pull/18301, that has most of those snapshots. Regardless, I added the snapshot for the chips file too.

File: ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      secretRecoveryPhrase: TEST_SEED.split(' '),
+    };
+
+    const { container } = renderWithProvider(
+      <RecoveryPhraseChips {...props} />,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
```
Comment (DDDDDanica) [inline ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.test.js:20]: I think if we test snapshot for `hideSeedPhrase` as `true` and `false`, will cover more situations. wdyt

File: ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    const props = {
+      secretRecoveryPhrase: TEST_SEED.split(' '),
+    };
+
+    const { container } = renderWithProvider(
+      <RecoveryPhraseChips {...props} />,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
```
Comment (DDDDDanica) [inline ui/pages/onboarding-flow/recovery-phrase/recovery-phrase-chips.test.js:19]: Also we can add simple tests to cover the condition of 
```
            confirmPhase &&
            indicesToCheck &&
            indicesToCheck.includes(index)
```

**Review by DDDDDanica — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**